creator-node/test/audiusUsers.js

```diff
@@ -66,3 +84,88 @@ describe('test AudiusUsers', function () {
       .expect(200)
   })
 })
+
+describe('tests /audius_users/metadata metadata upload with actual ipfsClient', function () {
+  let app, server, session, libsMock, ipfs
+
+  // Will need a '.' in front of storagePath to look at current dir
+  // a '/' will search the root dir
+  before(async () => {
+    const originalStoragePath = config.get('storagePath')
+    if (originalStoragePath.slice(0, 1) === '/') {
+      const updatedStoragePath = '.' + originalStoragePath
+      config.set('storagePath', updatedStoragePath)
+    }
+  })
+
+  beforeEach(async () => {
+    ipfs = ipfsClient.ipfs
+    libsMock = getLibsMock()
+
+    const appInfo = await getApp(ipfs, libsMock, BlacklistManager)
+    await BlacklistManager.blacklist(ipfs)
+
+    app = appInfo.app
+    server = appInfo.server
+    session = await createStarterCNodeUser()
+  })
+
+  afterEach(async () => {
+    sinon.restore()
+    await server.close()
+  })
+
+  it('should fail if metadatda is not found in request body', async function () {
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send({ dummy: 'data' })
+      .expect(500)
+
+    assert.deepStrictEqual(resp.body.error, 'Internal server error') // should be specific?
+  })
+
+  it('should throw error response if saving metadata to fails', async function () {
+    sinon.stub(ipfs, 'add').rejects(new Error('ipfs add failed!'))
+
+    const metadata = { metadata: 'spaghetti' }
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send(metadata)
+      .expect(500)
+
+    assert.deepStrictEqual(resp.body.error, 'Could not save file to disk, ipfs, and/or db: Error: ipfs add failed!')
+  })
+
+  it('successfully adds metadata file to filesystem, db, and ipfs', async function () {
+    const metadata = { spaghetti: 'spaghetti' }
+    const resp = await request(app)
+      .post('/audius_users/metadata')
+      .set('X-Session-ID', session)
+      .send({ metadata })
+      .expect(function (res) {
+        if (res.body.error) {
+          console.error(res.body.error)
+          assert.fail(res.body.error)
+        }
+      })
+      .expect(200)
+
+    // check that the metadata file was written to storagePath under its multihash
+    const metadataPath = path.join(config.get('storagePath'), resp.body.metadataMultihash)
+    assert.ok(fs.existsSync(metadataPath))
+
+    // check that the metadata file contents match the metadata specified
+    const metadataFileData = fs.readFileSync(metadataPath, 'utf-8')
+    assert.ok(metadataFileData, metadata)
```

Comment:
i don't think this does what you want in terms of checking equality

Reply:
ah, I meant to do assert.deepStrictEqual(). will update!